internal/api/block_documents.go

```diff
@@ -0,0 +1,142 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+type BlockDocumentClient interface {
+	Get(ctx context.Context, id uuid.UUID) (*BlockDocument, error)
+	Create(ctx context.Context, payload BlockDocumentCreate) (*BlockDocument, error)
+	Update(ctx context.Context, id uuid.UUID, payload BlockDocumentUpdate) error
+	Delete(ctx context.Context, id uuid.UUID) error
+
+	GetACL(ctx context.Context, id uuid.UUID) (*BlockDocumentAccess, error)
+	UpdateACL(ctx context.Context, id uuid.UUID, payload BlockDocumentAccessReplace) error
+}
+
+type BlockDocument struct {
+	BaseModel
+	Name *string `json:"name"` // names are optional for anonymous blocks
+	Data string  `json:"data"`
+
+	BlockSchemaID uuid.UUID `json:"block_schema_id"`
+	// BlockSchema   *BlockSchema `json:"block_schema"`
+
+	BlockTypeID   uuid.UUID `json:"block_type_id"`
+	BlockTypeName *string   `json:"block_type_name"`
+	BlockType     BlockType `json:"block_type"`
+
+	BlockDocumentReferences string `json:"block_document_references"`
+
+	IsAnonymous bool `json:"is_anonymous"`
+}
+
+type BlockDocumentCreate struct {
+	Name          *string   `json:"name"` // names are optional for anonymous blocks
+	Data          string    `json:"data"`
+	BlockSchemaID uuid.UUID `json:"block_schema_id"`
+	BlockTypeID   uuid.UUID `json:"block_type_id"`
+	IsAnonymous   bool      `json:"is_anonymous"`
+}
+
+type BlockDocumentUpdate struct {
+	BlockSchemaID     *uuid.UUID `json:"block_schema_id"`
+	Data              string     `json:"data"`
+	MergeExistingData bool       `json:"merge_existing_data"`
+}
+
+// AccessControlList is a custom type that represents
+// an API response where the value can be:
+// []uuid.UUID - a list of IDs
+// ["*"] - a wildcard, meaning "all".
+//
+// nolint:musttag // we have custom marshal/unmarshal logic for this type
+type AccessControlList struct {
+	IDs []uuid.UUID
+	All bool
+}
```

Comment:
basically, i've set up this struct type to allow us to serialize and better handle this varied type from the API

Reply:
Looks good, two quick questions:

- The API seems to refer to these as `access_control` objects. Would `type AccessControl struct` make more sense here? Maybe not, but it threw me off a bit when searching the API docs.
- Looks like `access_control` is part of the payload for [Set Deployment Access](https://app.prefect.cloud/api/docs#tag/Deployments/operation/set_deployment_access_api_accounts__account_id__workspaces__workspace_id__deployments__id__access_put) as well. Would it make sense to put this struct elsewhere, in a common/helpers file? (happy to merge as-is now and we can move it when we work on support for Deployments).

Reply:
> The API seems to refer to these as access_control objects. Would type AccessControl struct make more sense here? Maybe not, but it threw me off a bit when searching the API docs.

I'm down for that - although when I was skimming our API spec, i interpreted "access_control" = the entire payload, including the top-level keys for `manage`, `view`, etc., which each point to a list of IDs / `"*"`.  And then this here custom type would be for the weird, varied type holding the actual list of IDs (or the optional `"*"` string).  

> Looks like access_control is part of the payload for Set Deployment Access as well. Would it make sense to put this struct elsewhere, in a common/helpers file? (happy to merge as-is now and we can move it when we work on support for Deployments).

For sure. let me actually just create a helper module in `api` package and move this type there

Reply:
> I'm down for that - although when I was skimming our API spec, i interpreted "access_control" = the entire payload, including the top-level keys for manage, view, etc., which each point to a list of IDs / "*". And then this here custom type would be for the weird, varied type holding the actual list of IDs (or the optional "*" string).

Makes sense to me as well, happy to leave it as it is 👍🏼 

> For sure. let me actually just create a helper module in api package and move this type there

Sounds great.